cmds/boot/systemboot/main.go

```diff
@@ -208,14 +242,65 @@ func addSEL(sequence string) {
 	}
 }
 
+// getDebugEnabled checks whether debug output is requested, either via command line or via VPD
+// variables.
+// If -q was explicitly passed on the command line, will use that value, otherwise will look for
+// the VPD variable "systemboot_log_level".
+// Valid values are coreboot loglevels https://review.coreboot.org/cgit/coreboot.git/tree/src/commonlib/include/commonlib/loglevel.h,
+// either as integer (1, 2) or string ("debug").
+// If the VPD variable is missing or it is set to an invalid value, it will use the default.
+func getDebugEnabled() bool {
+	if isFlagPassed("q") {
+		return !*doQuiet
+	}
+
+	// -q was not passed, so `doQuiet` contains the default value
+	defaultDebugEnabled := !*doQuiet
+	// check for the VPD variable "systemboot_log_level". First the read-write, then the read-only
+	v, err := vpd.Get(vpdSystembootLogLevel, false)
+	if err != nil {
+		// TODO do not print warning if file is not found
+		log.Printf("Warning: failed to read read-write VPD variable '%s', will try the read-only one. Error was: %v", vpdSystembootLogLevel, err)
+		v, err = vpd.Get(vpdSystembootLogLevel, true)
+		if err != nil {
+			// TODO do not print warning if file is not found
+			log.Printf("Warning: failed to read read-only VPD variable '%s', will use the default value. Error was: %v", vpdSystembootLogLevel, err)
+			return defaultDebugEnabled
+		}
+	}
+	level := strings.ToLower(strings.TrimSpace(string(v)))
```

Comment:
I think you can skip a lot of this. 
I don't see the need to convert to a string (which you don't use?) when you can catch the case below?
switch (strings.TrimSpace(string(v))) {
case "whatrever"
case "0":
case "1":
etc.
or even 
case "0", "1", "2", "3": 
return false
case "4", etc.
default:
return defaultDebugEnabled

and so on. 
lines 271-280 seem unnecessary

Reply:
Hmm not sure I follow. I am using the string levels (in the switch).
I just translate integer levels (if any) to the equivalent string to make the switch afterwards more readable.

Reply:
Ah I got it now, changing